spec/lib/logging/third_party_transaction_spec.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Logging::ThirdPartyTransaction do
+  # to use shared examples
+  # 1. define variable test_object of desired type using a `let` block
+  # 2. test_object must define an instance method `method_to_wrap`
+  # 3. test_object should extend Logging::ThirdPartyTransaction::MethodWrapper
+  # 4. test_object definition should end with a call to .new, ensuring an object
+  #    is returned to your variable, rather than the class definition block,
+  #    as our linter will reject this.
+  shared_examples_for 'a third party transaction logger' do
+    context 'happy path' do
+      it 'wraps a method in logging actions' do
+        expect(test_object).to receive(:log_3pi_begin).at_least(:once)
+        expect(test_object).to receive(:log_3pi_complete).at_least(:once)
+
+        test_object.method_to_wrap
+      end
+
+      it 'returns the original methods return value' do
+        expect(test_object.method_to_wrap).to eq 'return value'
+      end
+    end
+
+    context 'when something goes wrong' do
+      it 'fails quietly and logs the problem' do
+        # Time is used in the logging methods
+        allow(Time).to receive(:current).and_raise(StandardError)
```

Comment:
(Blocking pending answer) Probably unlikely this will change but I'm not sure about this test picking one line of functionality within the method and mocking it failing. Is there an alternative approach/refactor that doesn't require this unit test knowing about a specific implementation detail of a method it is testing?

Reply:
[PENDING ACTION\] So the point of the test is to force a failure to ensure quite failure, so as far as the stated purpose, i think this checks the box, and i think it's an important box to check.

That being said, yes I also kinda hate this.  I spent about an hour trying to come up with a better way and couldn't...   Def open to suggestions, I can also take another pass at it. 🤔

Reply:
On further thought, I'm not sure there is a better way to do this.  In order to test this functionality we need an error to be raised from *within* the tested method.  I really can't think of any other way to do this, but like I said I'm open to suggestions ¯\_(  ͠◔__  ͡◔)_/¯